magda-authorization-api/src/createApiRouter.ts

```diff
@@ -33,6 +33,20 @@ export default function createApiRouter(options: ApiRouterOptions) {
             .then(() => res.end());
     }
 
+    router.all("/private/*", function(req, res, next) { //--- private API requires admin level access
+        
+        getUserIdHandling(req, res, options.jwtSecret, async (userId: string) => {
+            try{
+                const user = (await database.getUser(userId)).valueOrThrow(new Error(`Cannot locate user record by id: ${userId}`));
+                if(!user.isAdmin) throw new Error("Only admin users are authorised to access this API");
+                next();
+            }catch(e){
+                console.warn(e);
+                res.status(401).send("Not authorized");
```

Comment:
Right now this will return `401` for both non-existent user and existent user who isn't an admin - it'd be great if we could make it `403 Forbidden` for non-admins!

Reply:
Good idea~ Will make changes accordingly. I also noticed my changes impact `magda-admin-api` module. Will also make sure error codes are consistent in the test case tests of that modules as well.

Reply:
This has been done in commit 1e89cee6d87cc66b1f12c0a624e5215d822cad46